25_exceptions_lab/01.py

```diff
@@ -8,13 +8,19 @@
     numStr = raw_input('Enter a number:')
     
     try:
-        num = float(numStr)
-    except ValueError:
-        print ("%s is not a number\n" % numStr)
+        print math.sqrt(float(numStr))
+
+    except ValueError as e:
+        if e.message == 'math domain error':
+            print ("%s Number is not possitive\n" %numStr)
+        else:
+            print ("%s is not a number\n" % numStr)
+   
         exit(0)
     
     
-    if num < 0:
-        raise ValueError("Expected parameter >= 0, Got: %d" % num)
 
-    print math.sqrt(num)
+    
+    
+
+    

```

Comment:
👍 comparing the error string is usually not a good idea as that string might change in the future or it might depend on the current locale, but in this case you don't have a choice if you really want to give your own message instead of the e.message